### PR TITLE
Adding a changeThreadUserNickname

### PR DESCRIPTION
**Pull request by GAMELASTER**

### CODE REVIEW

File: src/changeThreadUserNickname.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function changeThreadUserNickname(nickname, threadID, participantID, callback) {
+		
+    var form = {
+      'nickname': nickname,
+			'participant_id': participantID,
```
Comment (bsansouci) [inline src/changeThreadUserNickname.js]: Can you fix the indentation?

File: src/changeThreadUserNickname.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      'thread_or_other_fbid': threadID
+    };
+		console.log(form);
```
Comment (bsansouci) [inline src/changeThreadUserNickname.js]: remove logging

File: src/changeThreadUserNickname.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    defaultFuncs
+      .post("https://www.messenger.com/messaging/save_thread_nickname/?source=thread_settings&dpr=1", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+				console.log(resData);
```
Comment (bsansouci) [inline src/changeThreadUserNickname.js]: Fix all indentation. We indent with 2 spaces (no tabs).

**Comment by bsansouci** [comment]
Thanks! @Schmavery and I will test this some time today.

**Comment by GAMELASTER** [comment]
Whoops! I got some problems with my work, I must deleted it so I pushed a old code. Now its good!

File: DOCS.md [hunk 4]
```diff
@@ -6,6 +6,7 @@
 * [`api.changeGroupImage`](#changeGroupImage)
 * [`api.changeThreadColor`](#changeThreadColor)
 * [`api.changeThreadEmoji`](#changeThreadEmoji)
+* [`api.changeThreadUserNickname`](#changeThreadUserNickname)
```
Comment (Schmavery) [inline DOCS.md]: Can we rename this to `changeNickname`?  I don't think there is any other kind.

File: src/changeThreadUserNickname.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      'participant_id': participantID,
+      'thread_or_other_fbid': threadID
+    };
+
+    defaultFuncs
+      .post("https://www.messenger.com/messaging/save_thread_nickname/?source=thread_settings&dpr=1", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+        if (resData.error === 1545014) {
+          throw {error: "Trying to change user nickname wasnt in thread"};
```
Comment (Schmavery) [inline src/changeThreadUserNickname.js]: I don't understand this error message.  Maybe you mean "Trying to change the nickname of a user not in this thread" or something similar?

File: src/changeThreadUserNickname.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        if (resData.error === 1357031) {
+          throw {error: "Trying to change user nickname of a chat that doesn't exist. Have at least one message in the thread before trying to change the user nickname."};
```
Comment (Schmavery) [inline src/changeThreadUserNickname.js]: Please replace "chat" with "thread"

File: src/changeThreadUserNickname.js (hunk 5 above)
Comment (GAMELASTER) [inline src/changeThreadUserNickname.js]: Yes, I changed that in new commit

File: src/changeThreadUserNickname.js (hunk 6 above)
Comment (GAMELASTER) [inline src/changeThreadUserNickname.js]: Done! Well, this error message is copied from the changeEmoji function, so its contains this naming too.

**Comment by bsansouci** [comment]
Ok seems to be working on my side. Thanks a ton!

**Pull request merged**